src/commons/editor/tabs/EditorTab.tsx

```diff
@@ -24,6 +25,10 @@ const EditorTab: React.FC<Props> = ({ filePath, isActive, setActive, remove }) =
       })}
       onClick={setActive}
     >
+      {' '}
+      {readOnly !== undefined && (
+        <Icon className="file-mode-button" icon={readOnly ? IconNames.LOCK : IconNames.UNLOCK} />
```

Comment:
Just a suggestion: the icon's meaning may not be clear to everyone. Perhaps we can add a tooltip as well?